mmv1/products/apigee/TargetServer.yaml

```diff
@@ -0,0 +1,154 @@
+    description: Specifies TLS configuration info for this TargetServer. The JSON name is sSLInfo for legacy/backwards compatibility reasons -- Edge originally supported SSL, and the name is still used for TLS configuration.
+    properties:
+      - !ruby/object:Api::Type::Boolean
+        name: 'enabled'
+        description: |
+          Enables TLS. If false, neither one-way nor two-way TLS will be enabled.
+        required: true
+      - !ruby/object:Api::Type::Boolean
+        name: 'clientAuthEnabled'
+        description: |
+          Enables two-way TLS.
+      - !ruby/object:Api::Type::String
+        name: 'keyStore'
+        description: |
+          Required if clientAuthEnabled is true. The resource ID of the keystore.
+      - !ruby/object:Api::Type::String
+        name: 'keyAlias'
+        description: |
+          Required if clientAuthEnabled is true. The resource ID for the alias containing the private key and cert.
+      - !ruby/object:Api::Type::String
+        name: 'trustStore'
+        description: |
+          The resource ID of the truststore.
+      - !ruby/object:Api::Type::Boolean
+        name: 'ignoreValidationErrors'
+        description: |
+          If true, Edge ignores TLS certificate errors. Valid when configuring TLS for target servers and target endpoints, and when configuring virtual hosts that use 2-way TLS. When used with a target endpoint/target server, if the backend system uses SNI and returns a cert with a subject Distinguished Name (DN) that does not match the hostname, there is no way to ignore the error and the connection fails.
+      - !ruby/object:Api::Type::Array
+        name: 'protocols'
+        item_type: Api::Type::String
+        description: |
+          The TLS versioins to be used.
+      - !ruby/object:Api::Type::Array
+        name: 'ciphers'
+        item_type: Api::Type::String
+        description: |
+          The SSL/TLS cipher suites to be used. For programmable proxies, it must be one of the cipher suite names listed in: http://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#ciphersuites. For configurable proxies, it must follow the configuration specified in: https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#Cipher-suite-configuration. This setting has no effect for configurable proxies when negotiating TLS 1.3.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'commonName'
+        description: The TLS Common Name of the certificate.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'value'
+            description: |
+              The TLS Common Name string of the certificate.
+          - !ruby/object:Api::Type::Boolean
+            name: 'wildcardMatch'
+            description: |
+              Indicates whether the cert should be matched against as a wildcard cert.
+
+  - !ruby/object:Api::Type::Enum
+    name: 'protocol'
+    immutable: true
+    default_value: :HTTP
```

Comment:
From [the docs about this field](https://cloud.google.com/apigee/docs/reference/apis/apigee/rest/v1/organizations.environments.targetservers?hl=en#Protocol) I can see that:

> UNSPECIFIED defaults to HTTP for backwards compatibility

I think instead of setting a default here, we should let the API supply the default and have the provider pull that value from the API using `default_from_api: true` ([see here](https://github.com/GoogleCloudPlatform/magic-modules/blob/f000b0d242218c0b7f153fa643ab368996e008e8/docs/content/develop/resource.md?plain=1#L273-L283)). This means a user can still get a usable resource while leaving the protocol field unset, but also allows the provider to adapt to the API changing the default value. Does that sound ok?

Reply:
Agreed. Change checked in.